gpcontrib/arenadata_toolkit/sql/upgrade_test.sql

```diff
@@ -0,0 +1,145 @@
+CREATE FUNCTION do_upgrade_test_for_arenadata_toolkit(from_version TEXT)
+RETURNS setof TEXT
+AS $$
+BEGIN
+
+-- Simple check: only create and alter extension:
+	CREATE EXTENSION arenadata_toolkit VERSION '1.0';
+	IF from_version != '1.0'
+	THEN
+		EXECUTE FORMAT($fmt$ALTER EXTENSION arenadata_toolkit
+							UPDATE TO %1$I;$fmt$, from_version);
+	END IF;
+	ALTER EXTENSION arenadata_toolkit UPDATE;
+
+-- Check the result
+	IF (SELECT default_version = installed_version
+		FROM pg_available_extensions
+		WHERE name='arenadata_toolkit')
+	THEN
+		RETURN NEXT from_version || ': only alter check';
+	END IF;
+
+-- Cleanup before next step
+	DROP EXTENSION arenadata_toolkit;
+	DROP SCHEMA arenadata_toolkit CASCADE;
+
+-- Create, adb_create_tables and alter extension:
+	CREATE EXTENSION arenadata_toolkit VERSION '1.0';
+	IF from_version != '1.0'
+	THEN
+		EXECUTE FORMAT($fmt$ALTER EXTENSION arenadata_toolkit
+							UPDATE TO %1$I;$fmt$, from_version);
+	END IF;
+	PERFORM arenadata_toolkit.adb_create_tables();
+	ALTER EXTENSION arenadata_toolkit UPDATE;
+
+-- Check the result
+	IF (SELECT default_version = installed_version
+		FROM pg_available_extensions
+		WHERE name='arenadata_toolkit')
+	THEN
+		RETURN NEXT from_version || ': alter and create_tables check';
+	END IF;
+
+-- Cleanup before next step
+	DROP EXTENSION arenadata_toolkit;
+	DROP SCHEMA arenadata_toolkit CASCADE;
+
+-- Create, adb_create_tables, adb_collect_table_stats and alter extension:
+	CREATE EXTENSION arenadata_toolkit VERSION '1.0';
+	IF from_version != '1.0'
+	THEN
+		EXECUTE FORMAT($fmt$ALTER EXTENSION arenadata_toolkit
+						UPDATE TO %1$I;$fmt$, from_version);
+	END IF;
+	PERFORM arenadata_toolkit.adb_create_tables();
+	PERFORM arenadata_toolkit.adb_collect_table_stats();
+	ALTER EXTENSION arenadata_toolkit UPDATE;
+
+-- Check the result
+	IF (SELECT default_version = installed_version
+		FROM pg_available_extensions
+		WHERE name='arenadata_toolkit')
+	THEN
+		RETURN NEXT from_version || ': alter, create_tables and collect_table_stats check';
+	END IF;
+
+-- Check field "tablespace_location" and table "db_files_history_backup_YYYYMMDDtHHMMSS"
+-- which were added at version 1.4
+	PERFORM arenadata_toolkit.adb_create_tables();
+
+	IF 4 = (SELECT count(1)
+			FROM pg_class c
+			LEFT JOIN pg_namespace n ON n.oid = c.relnamespace
+			LEFT JOIN pg_attribute a ON a.attrelid = c.oid AND
+			                            a.attname = 'tablespace_location'
+			WHERE n.nspname = 'arenadata_toolkit' AND
+			      c.relname IN ('db_files_current',
+			                    '__db_files_current',
+			                    '__db_files_current_unmapped',
+			                    'db_files_history') AND
+			      a.attrelid IS NOT NULL)
+	THEN
+		RETURN NEXT from_version || ': column tablespace_location check';
+	END IF;
+
+	IF EXISTS (SELECT 1
+			   FROM pg_tables
+			   WHERE schemaname='arenadata_toolkit' AND
+			         tablename SIMILAR TO 'db_files_history_backup_[0-9]{8}t[0-9]{6}')
```

Comment:
What will be when we update extension to 1.5? 
We will execute do_updrade_test for from_version 1.4(otherwise what's the point of the whole update test), 
and what will be with that check?

Reply:
At this case backup is not needed (field `tablespace_location` has already existed at table `db_files_history`). When we execute `SELECT do_upgrade_test_for_arenadata_toolkit('1.0') ORDER BY 1;`, we will not show line : "1.4:  db_files_history_backup check". I think, it is normal case, we can check, that extra actions are not done.